crates/ruma-macros/src/lib.rs

```diff
@@ -108,27 +215,194 @@ pub fn event_enum(input: TokenStream) -> TokenStream {
+/// {Kind}Event<FooEventContent>`.
+///
+/// The only exception for this is if the type has the `GlobalAccountData + RoomAccountData` kinds,
+/// it generates two type aliases:
+///
+/// * `type GlobalFooEvent = GlobalAccountDataEvent<FooEventContent>`
+/// * `type RoomFooEvent = RoomAccountDataEvent<FooEventContent>`
+///
+/// Some kinds generate more type aliases:
+///
+/// * `type SyncFooEvent = Sync{Kind}Event<FooEventContent>`: an event received via the `/sync` API,
+///   for the `MessageLike`, `State` and `Ephemeral` kinds
+/// * `type OriginalFooEvent = Original{Kind}Event<FooEventContent>`, a non-redacted event, for the
+///   `MessageLike` and `State` kinds
+/// * `type OriginalSyncFooEvent = OriginalSync{Kind}Event<FooEventContent>`, a non-redacted event
+///   received via the `/sync` API, for the `MessageLike` and `State` kinds
+/// * `type RedactedFooEvent = Redacted{Kind}Event<RedactedFooEventContent>`, a redacted event, for
+///   the `MessageLike` and `State` kinds
+/// * `type OriginalSyncFooEvent = RedactedSync{Kind}Event<RedactedFooEventContent>`, a redacted
+///   event received via the `/sync` API, for the `MessageLike` and `State` kinds
+/// * `type InitialFooEvent = InitialStateEvent<FooEventContent>`, an event sent during room
+///   creation, for the `State` kind
+/// * `type StrippedFooEvent = StrippedStateEvent<PossiblyRedactedFooEventContent>`, an event that
+///   is in a room state preview when receiving an invite, for the `State` kind
 ///
 /// You can use `cargo doc` to find out more details, its `--document-private-items` flag also lets
 /// you generate documentation for binaries or private parts of a library.
 ///
-/// By default, the type this macro is used on and the generated types get a `#[non_exhaustive]`
-/// attribute. This behavior can be controlled by setting the `ruma_unstable_exhaustive_types`
-/// compile-time `cfg` setting as `--cfg=ruma_unstable_exhaustive_types` using `RUSTFLAGS` or
-/// `.cargo/config.toml` (under `[build]` -> `rustflags = ["..."]`). When that setting is
-/// activated, the attribute is not applied so the types are exhaustive.
+/// # Syntax
+///
+/// The basic syntax for using this macro is:
+///
+/// ```ignore
+/// #[derive(Clone, Debug, Deserialize, Serialize, EventContent)]
+/// #[ruma_event(type = "m.foo_bar", kind = MessageLike)]
+/// pub struct FooBarEventContent {
+///     data: String,
+/// }
+/// ```
+///
+/// ## Container attributes
+///
+/// The following settings can be used on the container, with the `#[ruma_event(_)]` attribute.
+/// `type` and `kind` are always required.
+///
+/// ### `type = "m.event_type"`
+///
+/// The `type` of the event according to the Matrix specification, always required. This is usually
+/// a string with an `m.` prefix.
+///
+/// Types with an account data kind can also use the `.*` suffix, if the end of the type changes
+/// dynamically. It must be associated with a field that has the `type_fragment` attribute that will
```

Comment:
`type_fragment` is also nested under `ruma_event` like other things, right?

Reply:
Yes I added the rest of the attribute.